src/firefly/java/edu/caltech/ipac/firefly/data/sofia/DataExtractUtil.java

```diff
@@ -0,0 +1,47 @@
+package edu.caltech.ipac.firefly.data.sofia;
+
+import edu.caltech.ipac.firefly.data.ServerRequest;
+import edu.caltech.ipac.firefly.server.ServerCommandAccess;
+import edu.caltech.ipac.firefly.server.ServerContext;
+import edu.caltech.ipac.firefly.server.SrvParam;
+import edu.caltech.ipac.firefly.server.cache.UserCache;
+import edu.caltech.ipac.firefly.server.query.EmbeddedDbProcessor;
+import edu.caltech.ipac.firefly.server.util.multipart.UploadFileInfo;
+import edu.caltech.ipac.table.DataGroup;
+import edu.caltech.ipac.util.cache.Cache;
+import edu.caltech.ipac.util.cache.StringKey;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+
+/**
+ * Should implement child that returns datagroups,
+ * source file from request or overwritten in child or unit/integration tests
+ * Could be called as link service or return a datalink table
+ */
+public abstract class DataExtractUtil {
```

Comment:
Util classes should be written as public static.  If this is intended to be a DataExtractor base class or interface, then it should be promoted outside of sofia and have well-defined usage.